test-data/unit/check-literal.test

```diff
@@ -2945,3 +2945,88 @@ class C(B[Literal["word"]]):
 reveal_type(C().collection)  # N: Revealed type is "builtins.list[Literal['word']]"
 reveal_type(C().word)  # N: Revealed type is "Literal['word']"
 [builtins fixtures/tuple.pyi]
+
+[case testLiteralNarrowingInExpr]
+from typing import Any
+from typing_extensions import Literal
+
+def narrow(x: int, y: str, z: Literal[1, 2, 3]):
+    if x in (1, 2, "a", "b"):
```

Comment:
Ideally I'd like to also test that this works for `list` constructors (`[1, 2]`) and `set` constructors (`{"a", "b"}`) but could not figure out how to get hte fixtures to play nicely with that.

Reply:
As a relatively uninformed outsider - if you were to adjust this test to use `if x == 1 or x == 2 or x == "a" or x == "b":` will it narrow identically to what you have here?

Reply:
Wouldn't you agree that inconsistent narrowing semantics with `==` could be confusing to users of Mypy?

As an even simpler example:

```python
def test_narrowing(x: int) -> None:
    if x in (1,):
        reveal_type(x)  # note: Revealed type is "builtins.int" (but presumably would be "Literal[1] after this PR)
    
    if x == 1:
        reveal_type(x)  # note: Revealed type is "builtins.int"
```

For what its' worth, Pyright does this correctly and consistently:

```python
def test_narrowing(x: int) -> None:
    if x in (1,):
        reveal_type(x)  # Type of "x" is "Literal[1]"
    
    if x == 1:
        reveal_type(x)  # Type of "x" is "Literal[1]"
```

Reply:
> Wouldn't you agree that inconsistent narrowing semantics with == could be confusing to users of Mypy?

Not particularly, no. It's just a missing feature like any other. I'm not saying that "someone" shouldn't do this, but I'm saying that *I'm* not planning on doing it in this PR. If you see an easy way of adding support, then feel free to pointers in the code!